packages/did/src/DidDetails/LightDidDetails.utils.ts

```diff
@@ -82,20 +90,79 @@ export function serializeAndEncodeAdditionalLightDidDetails({
     return null
   }
 
-  return cborEncode(objectToSerialize).toString('base64')
+  return cborEncode(objectToSerialize)
+}
+
+/**
+ * Serialize the optional encryption key of an off-chain DID using the CBOR serialization algorithm and encoding the result in the specified format, if supported.
+ *
+ * @param details The light DID details to encode.
+ * @param details.encryptionKey The DID encryption key.
+ * @param details.serviceEndpoints The DID service endpoints.
+ * @param details.detailsEncoding The algorithm to use to encode a light DID details. Base58 is the default and recommended.
+ * @returns The encoded and CBOR-serialized off-chain DID optional details.
+ */
+export function serializeAndEncodeAdditionalLightDidDetails({
+  encryptionKey,
+  serviceEndpoints,
+  detailsEncoding,
+}: Pick<
+  LightDidDetailsCreationOpts,
+  'encryptionKey' | 'serviceEndpoints' | 'detailsEncoding'
+>): string | null {
+  const serialisedObject = serializeAdditionalLightDidDetails({
+    encryptionKey,
+    serviceEndpoints,
+  })
+  if (!serialisedObject) {
+    return null
+  }
+  if (detailsEncoding === 'base64') {
+    return base64Encode(serialisedObject)
+  }
+  if (detailsEncoding === 'base58') {
+    return base58Encode(serialisedObject)
+  }
+  throw new Error(
+    `Specified encoding ${detailsEncoding} not supported or undefined.`
+  )
+}
+
+export type DetailsDecodingResult = Pick<
+  LightDidDetailsCreationOpts,
+  'encryptionKey' | 'serviceEndpoints'
+> & {
+  detailsEncoding: 'base58' | 'base64'
 }
 
+/**
+ * Decode and deserialize encryption key and service endpoints from the provided encoded representation.
+ *
+ * @param rawInput The encoded string for the details to decode.
+ * @param version The version of the light DID to use to decode the details, if present. It defaults to the latest supported version.
+ * @returns The result of the decoding operation.
+ */
 export function decodeAndDeserializeAdditionalLightDidDetails(
   rawInput: string,
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  version = 1
-): Pick<LightDidDetailsCreationOpts, 'encryptionKey' | 'serviceEndpoints'> {
-  const decodedPayload: Map<string, unknown> = cborDecode(rawInput, {
-    encoding: 'base64',
-  })
+  version = LightDidDetails.LIGHT_DID_LATEST_VERSION
+): DetailsDecodingResult {
+  let deserialized: Uint8Array
+  let encoding: 'base58' | 'base64'
+  try {
+    // Try decoding details as base58
+    deserialized = base58Decode(rawInput)
```

Comment:
Since there can be collisions between base64 and base58 encoding, this might not throw on a base64 string.

Reply:
I did dig deeper into this and it turns out that since we are always including `{ e: { publickey: [...], type: "..." }}`, base64 encoded string always starts with `oWFlomlwdWJsaWNrZXnYQFhA` and ends with `R0eXBlZ3NyMjU1MTk=`, which include characters not allowed in base58.
So it would actually work, but might not be the best solution here.
We should maybe think about something like https://github.com/multiformats/multibase

Reply:
Also we should think about reducing the characters which are the same for every encoded string